pkg/appgateserver/cmd/cmd.go

```diff
@@ -0,0 +1,199 @@
+package cmd
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"log"
+	"net/http"
+	"net/url"
+	"os"
+	"os/signal"
+
+	"cosmossdk.io/depinject"
+	ring_secp256k1 "github.com/athanorlabs/go-dleq/secp256k1"
+	ringtypes "github.com/athanorlabs/go-dleq/types"
+	cosmosclient "github.com/cosmos/cosmos-sdk/client"
+	"github.com/cosmos/cosmos-sdk/client/flags"
+	"github.com/cosmos/cosmos-sdk/crypto/keyring"
+	"github.com/cosmos/cosmos-sdk/crypto/keys/secp256k1"
+	cryptotypes "github.com/cosmos/cosmos-sdk/crypto/types"
+	"github.com/spf13/cobra"
+
+	"github.com/pokt-network/poktroll/pkg/appgateserver"
+	blockclient "github.com/pokt-network/poktroll/pkg/client/block"
+	eventsquery "github.com/pokt-network/poktroll/pkg/client/events_query"
+)
+
+var (
+	signingKeyName    string
+	selfSigning       bool
+	listeningEndpoint string
+	cometWebsocketUrl string
+)
+
+func AppGateServerCmd() *cobra.Command {
+	cmd := &cobra.Command{
+		Use:   "appgate-server",
+		Short: "Starts the AppGate server",
+		Long: `Starts the AppGate server that will listen for incoming relays requests and will handle
+the interaction with the chain, sessions and suppliers in order to receive the correct
+response for the request.
+
+If the server is started with a defined --self-signing flag, it will behave
+as an application and sign any incoming requests with the private key associated with
+the --signing-key-name flag. If however, this flag is not provided, the server will
+behave as a gateway and will sign relays on behalf of any application sending it relays provided
+that the address recieved in the query parameters of a request has been delegated to by the
+gateway, this is so that it can sign relays using the ring of the application with the
+key associated with the --signing-key-name flag.
+
+If an application doesn't provide the --self-signing flag, it will be able to send relays
+to the AppGate server and it will still function as an application, however each request
+will have to contain the "?senderAddress=[address]" query parameter, where [address] is
+the address of the application that is sending the request. This is so that the server
+can generate the correct ring for the application and sign the request.`,
```

Comment:
```suggestion
		Long: `Starts the AppGate server that listens for incoming relay requests and handles the necessary on-chain interactions (sessions, suppliers, etc) to receive the respective relay response.

-- App Mode (Flag)- - 
If the server is started with a defined `--signing-key-name` flag, it will behave
as an Application. Any incoming requests will be signed with the Application's private key.

-- Gateway Mode (Flag)-- 
If the `--signing-key-name` flag is not provided, the server will behave as a Gateway. It will
sign relays on behalf of any Application sending it relays, provided that the address associated
with `--signing-key-name` has been delegated to. This is necessary for the application<->gateway
ring signature to function.

-- App Mode (HTTP) --
If an application doesn't provide the `--signing-key-name` flag, it can stillsend relays
to the AppGate server and function as an Application, provided that:
1. Each request contains the `?senderAddress=[address]` query parameter
2. The keyring on the AppGate server contains the privateKey corresponding to the address provided`
```

Reply:
Updated the comment to match the logic but kept the formatting